projects/igniteui-angular/src/lib/directives/text-highlight/text-highlight.directive.ts

```diff
@@ -335,7 +365,26 @@ export class IgxTextHighlightDirective implements AfterViewInit, AfterViewChecke
         const column = group.columnIndex === undefined ? group.column : group.columnIndex;
         const row = group.rowIndex === undefined ? group.row : group.rowIndex;
 
-        if (column === this.column && row === this.row && group.page === this.page) {
+        let metadataMatch = true;
+
+        if (group.metadata) {
+            if (group.metadata.size !== this.metadata.size) {
+                metadataMatch = false;
+            } else {
+                group.metadata.forEach((value, key) => {
+                    if (!metadataMatch) {
+                        return;
+                    }
+                    if (this.metadata.has(key)) {
+                        metadataMatch = this.metadata.get(key) === value;
+                    } else {
+                        metadataMatch = false;
+                    }
+                });
```

Comment:
Returning in the forEach won't break the loop. I would suggest to use `for of` instead so you can break the loop when metadataMatch is false.

Reply:
I would also suggest to extract this as a generic helper method for comparing maps in the utils file since almost the same code is repeated in the `compareMetadata` function.

Reply:
The use of `for of` would result in a build error, which prompts to the need to add a `downlevelIteration: true` to the tsconfig.json file, while targeting es5.